packages/zipkin/src/tracer/index.js

```diff
@@ -69,6 +69,17 @@ class Tracer {
     return childId;
   }
 
+  local(operationName, callable) {
+    return this.scoped(() => {
+      const traceId = this.createChildId();
+      this.setId(traceId);
+      this.recordAnnotation(new Annotation.LocalOperationStart(operationName));
+      const result = callable();
```

Comment:
What about when the capable is a) no callable? b) throws an error?

Reply:
I guess we can assume that it's a callable, what else should it be?
But it could potentially return a Promise, in which case we should not annotate LocalOperationStop until the Promise is finished. (Async function with try-catch-finally could work, with `const result = await callable();`)

Reply:
thanks for the tips! I added a commit. I do need some advice on how to handle not-callable cc also @pbadenski as we need whatever we do to make sense in typescript (not sure you can overload functions in ts or not)

Reply:
function overloading works in typescript, very natural too:
```
local<V>(name: string, callback: () => V): V;
local<V>(name: string, promise: Promise<V>): Promise<V>;
```

Reply:
```
local<V>(name: string, promise: Promise<V>): Promise<V>;
```
No no no, you cannot pass in a promise like that, because that would mean you would have to start running your work before we have added the "before" annotations in Zipkin. It has to be

```
local<V>(name: string, callback: () => Promise<V>): Promise<V>;
```

then zipkin-js can decide when to start the work (typically immediately after adding its "before" annotations).

There is really no point of handling whether or not `callback` is callable or not, there is no other input than a callback that makes sense. So if we get something that is not callable, we should just throw a TypeError, I think.